webapi/src/main/scala/org/knora/webapi/responders/v2/ontology/Cardinalities.scala

```diff
@@ -521,30 +529,58 @@ object Cardinalities {
     internalOntologyIri: SmartIri
   )(implicit ec: ExecutionContext): Future[Boolean] = {
     val currentOntologyState: ReadOntologyV2 = cacheData.ontologies(internalOntologyIri)
-    val currentClassState: ClassInfoContentV2 = currentOntologyState.classes
+
+    val readClassInfo: ReadClassInfoV2 = currentOntologyState.classes
       .getOrElse(
         internalClassIri,
         throw BadRequestException(
-          s"Class ${internalClassIri} does not exist"
+          s"Class $internalClassIri does not exist"
         )
       )
-      .entityInfoContent
-    val existingCardinality = currentClassState.directCardinalities
-      .getOrElse(
-        propertyIri,
+
+    // if cardinality is inherited, it's not directly defined on that class
+    if (readClassInfo.inheritedCardinalities.keySet.contains(propertyIri)) {
+      return FastFuture.successful(false)
+    }
+
+    val currentClassState: ClassInfoContentV2 = readClassInfo.entityInfoContent
+    val existingCardinality = currentClassState.directCardinalities.get(propertyIri)
+    existingCardinality match {
+      case Some(cardinality) =>
+        if (cardinality.cardinality.equals(cardinalityInfo.cardinality)) {
+          FastFuture.successful(true)
+        } else {
+          FastFuture.failed(
+            throw BadRequestException(
+              s"Submitted cardinality for property $propertyIri does not match existing cardinality."
+            )
+          )
+        }
+      case None =>
         throw BadRequestException(
-          s"Cardinality for property ${propertyIri} is not defined."
+          s"Submitted cardinality for property $propertyIri is not defined for class $internalClassIri."
```

Comment:
Maybe you could add a hint to the error message, that it's defined on the superclass?

Reply:
At that point, there could be other reasons why it is not defined (p.ex. just forgotten or so).